### ISSUE TITLE
More tests, more docs

### ISSUE THREAD

**Issue by fantix** [zeromq/zmq.rs#12]

**Comment by cjdsellers** [comment]
Hi all,

If I was to get involved with the project - are there any suggestions for where adding unit tests would be most valuable?

**Comment by TheButlah** [comment]
@cjdsellers integration tests against the reference zeromq implementation would be very valuable - specifically making sure that the library is compliant with the spec.

**Comment by cjdsellers** [comment]
Ok understood. I've working with ZMQ before in both C# and Python and integrated services between those languages. I'll familiarize myself with the spec some more.

**Comment by Alexei-Kornienko** [comment]
What I'm actually doing when working on some new functionality:
* Implement this code in Python using official driver.
* Replace some of the code with Rust equivalent. 
* Replace other parts step by step.

Maybe we could do the same for integrations tests so for example test Rust implementation against some python code (cause it requires only a few lines of code). Need to think about how this can be run locally and on CI. Possibly in a docker container?

**Comment by TheButlah** [comment]
@Alexei-Kornienko its probably fine to just rely on the `zmq` crate in our integration tests for now, since that is just a binding for libzmq

**Comment by Alexei-Kornienko** [comment]
@TheButlah yeah but still it's a bit harder to setup and configure.. maybe it's faster however for me Python is much easier to bootstrap.
